### PR TITLE
Split some modules (dummy & mint) into subcrates

### PR DESCRIPTION
**Pull request by dpc**
On top #1805 

Gradually removing client dependency on server side module code.

### CODE REVIEW

File: client/cli/Cargo.toml [hunk 1]
```diff
@@ -20,7 +20,8 @@ lightning-invoice = { version = "0.21.0", features = [ "serde" ] }
 mint-client = { path = "../client-lib" }
 fedimint-core ={ path = "../../fedimint-core" }
 fedimint-rocksdb = { path = "../../fedimint-rocksdb" }
-fedimint-mint = { path = "../../modules/fedimint-mint" }
+fedimint-mint-client = { path = "../../modules/fedimint-mint-client" }
```
Comment (dpc) [inline client/cli/Cargo.toml:23]: We only depend on client side here, yay!

File: client/client-lib/Cargo.toml [hunk 2]
```diff
@@ -64,3 +64,4 @@ test-log = { version = "0.2", features = [ "trace" ], default-features = false }
 tracing-subscriber = { version = "0.3.16", features = [ "env-filter" ] }
 fedimint-testing = { path = "../../fedimint-testing" }
 once_cell = "1.16.0"
+fedimint-mint-server = { path = "../../modules/fedimint-mint-server" }
```
Comment (dpc) [inline client/client-lib/Cargo.toml:67]: dev-deps only, as tests use it.

File: fedimintd/Cargo.toml [hunk 3]
```diff
@@ -41,7 +41,7 @@ fedimint-rocksdb = { path = "../fedimint-rocksdb" }
 fedimint-server = { path = "../fedimint-server" }
 fedimint-logging = { path = "../fedimint-logging", features = ["telemetry"] }
 fedimint-wallet = { path = "../modules/fedimint-wallet", features = ["native", "server"] }
-fedimint-mint= { path = "../modules/fedimint-mint", features = ["server"] }
+fedimint-mint-server = { path = "../modules/fedimint-mint-server" }
```
Comment (dpc) [inline fedimintd/Cargo.toml:44]: No more conditional compilation.

**Review by jkitman — APPROVED** [review]
So does `fedimint-server` no longer depend on `fedimint-mint-client`?

Will this be possible with to perform on the other modules as well?

**Comment by dpc** [comment]
> So does fedimint-server no longer depend on fedimint-mint-client?

It does, but that doesn't bother me. If fedimint-server needs to use client-side calls, then it can depend on client module.

It's the other way around that is more of a problem.

> Will this be possible with to perform on the other modules as well?

That's the plan, but I rather rebase smaller commits than one large one.

**Review by elsirion — APPROVED** [review]

**Comment by elsirion** [comment]
Great work, that will come in handy in #1447.

**Pull request merged**